src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationExtensions.cs

```diff
@@ -0,0 +1,91 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Dapr.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.DaprSecretStore;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Extension methods for registering <see cref="DaprSecretStoreConfigurationProvider"/> with <see cref="IConfigurationBuilder"/>.
+    /// </summary>
+    public static class DaprSecretStoreConfigurationExtensions
+    {
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Dapr Secret Store.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="store">Dapr secret store name.</param>
+        /// <param name="secrets">The secrets to retrieve.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddDaprSecretStore(
+            this IConfigurationBuilder configurationBuilder,
+            string store,
+            IEnumerable<DaprSecretDescriptor> secrets)
+        {
+            return AddDaprSecretStore(
+                configurationBuilder,
+                store,
+                secrets,
+                (builder) => { });
+        }
+
+        /// <summary>
+        /// Adds an <see cref="IConfigurationProvider"/> that reads configuration values from the Dapr Secret Store.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="store">Dapr secret store name.</param>
+        /// <param name="secrets">The secrets to retrieve.</param>
+        /// <param name="builder">Configures the Dapr client builder</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddDaprSecretStore(
+            this IConfigurationBuilder configurationBuilder,
+            string store,
+            IEnumerable<DaprSecretDescriptor> secrets,
+            Action<DaprClientBuilder> builder)
+        {
+            if (store == null)
+            {
+                throw new ArgumentNullException(nameof(store));
+            }
+
+            if (secrets == null)
+            {
+                throw new ArgumentNullException(nameof(secrets));
+            }
+
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            var daprClientBuilder = new DaprClientBuilder();
```

Comment:
I would go back to a method that receives the client:

```csharp
public static IConfigurationBuilder AddDaprSecretStore(
            this IConfigurationBuilder configurationBuilder,
            string store,
            IEnumerable<DaprSecretDescriptor> secretDescriptors,
            DaprClient client)
```

Removing:

```csharp
public static IConfigurationBuilder AddDaprSecretStore(
            this IConfigurationBuilder configurationBuilder,
            string store,
            IEnumerable<DaprSecretDescriptor> secretDescriptors)
```

This will work just like the `AddAzureKeyVault` extension works:

```csharp
public static IHostBuilder CreateHostBuilder(string[] args) =>
             Host.CreateDefaultBuilder(args)
                .ConfigureAppConfiguration((ctx, builder) =>
                {
                    var keyVaultEndpoint = GetKeyVaultEndpoint();
                    if (!string.IsNullOrEmpty(keyVaultEndpoint))
                    {
                        var azureServiceTokenProvider = new AzureServiceTokenProvider();
                        var keyVaultClient = new KeyVaultClient(
                            new KeyVaultClient.AuthenticationCallback(
                                azureServiceTokenProvider.KeyVaultTokenCallback));
                        builder.AddAzureKeyVault(
                            keyVaultEndpoint, keyVaultClient, new DefaultKeyVaultSecretManager());
                    }
                })
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStartup<Startup>();
                });
```

Reply:
@cmendible Do you think creating the client earlier like you were doing earlier, but only do it when its not in the DI container, when you create it add to DI container,  this would stop multiple client creations and users also dont have to pass it?

Reply:
I don't see how we can check if the Dapr client is already registered in DI without adding an an `IServiceCollection` param or an `IServiceProvider ` or `IApplicationBuilder` to get the instance if already present, which in my opinion would make the method far more complicated than needed.

Receiving the client will allow the following scenario:

```cshap
public static IHostBuilder CreateHostBuilder(string[] args)
        {
            DaprClient client = new DaprClientBuilder().Build();

            return Host.CreateDefaultBuilder(args)
                    .ConfigureServices((services) =>
                    {
                        services.AddSingleton<DaprClient>(client);
                    })
                    .ConfigureAppConfiguration((configBuilder) =>
                    {
                        var secretDescriptors = new DaprSecretDescriptor[] { new DaprSecretDescriptor("secretName") };
                        configBuilder.AddDaprSecretStore("store", secretDescriptors, client);
                    })
                    .ConfigureWebHostDefaults(webBuilder =>
                    {
                        webBuilder.UseStartup<Startup>();
                    });
        }
```

Reply:
@cmendible  This looks good to me.

Reply:
Fixed in 9d63986